action.yml

```
@@ -0,0 +1,140 @@
+name: Gauge release
+description: Gauge if should do a patch release
+
+inputs:
+  latest_local_sha:
+    description: The latest local sha
+    required: true
+    type: string
+
+outputs:
+  do_release:
+    description: Whether to do a release
+    value: ${{ steps.gauge-release.outputs.do_release }}
+  next_tag:
+    description: The patch tag to release
+    value: ${{ steps.gauge-release.outputs.next_tag }}
+
+runs:
+  using: composite
+  steps:
+    - name: Gauge release
+      id: gauge-release
+      shell: bash
+      env:
+        GITHUB_REPOSITORY: ${{ github.repository }}
+        GITHUB_REF_NAME: ${{ github.ref_name }}
+        GITHUB_SHA: ${{ github.sha }}
+        LATEST_LOCAL_SHA: ${{ inputs.latest_local_sha }}
+      run: |
+        DO_RELEASE=1
+
+        # Double check that LATEST_LOCAL_SHA matches GITHUB_SHA
+        if [[ $DO_RELEASE == "1" ]]; then
+          if [[ $LATEST_LOCAL_SHA != $GITHUB_SHA ]]; then
+            echo "LATEST_LOCAL_SHA $LATEST_LOCAL_SHA does not match GITHUB_SHA $GITHUB_SHA"
```

```suggestion
            echo "Not patch releasing because GITHUB_SHA is not equal to latest local sha"
```
This is clearer as it explicitly explains we're not doing the release, and this is the reason.
You can add an output at the top with what the two SHA values are if you like to aid debugging. That tends to be the way we provide this sort of information in our actions.

---

Updated